src/librustdoc/html/static/css/rustdoc.css

```diff
@@ -2743,17 +2748,19 @@ Original by Dempfi (https://github.com/dempfi/ayu)
 	border-right: 1px solid #ffb44c;
 }
 
-:root[data-theme="ayu"] .search-results a:hover,
-:root[data-theme="ayu"] .search-results a:focus {
+:root[data-theme="ayu"] .search-results li:hover > a,
```

Comment:
Having this special case for the ayu theme is really not great, we should try to either make a new variable or "fix" the ayu theme at some point (not for this PR).